File: lib/distance_between_points.rb
def distance(point_A, point_B) # Distance between two points (pairs of X,Y coordinates)
  x_diff = (point_A.first - point_B.first).abs
  y_diff = (point_A.last - point_B.last).abs
    if x_diff == 0 || y_diff == 0 # If only one axis differs simply calculate the difference
  	  dist = (x_diff + y_diff).round(3) # Rounded to 3 decimal places
    else # If both axis differ, use pythagorean formula:
      dist_squared = (x_diff ** 2) + (y_diff ** 2) # a^2 + b^2 = c^2 
      dist = sprintf('%.3f', Math.sqrt(dist_squared)).to_f # Take square root of c
  end
end

puts distance([1,3], [5,19])

# .sprintf is nice for formatting but it outputs a string, so you need to convert.




   
  

File: lib/distance_between_points.js
var distance = function(pointA, pointB){
  var xDiff = Math.abs(pointA[0] - pointB[0]);
  var yDiff = Math.abs(pointA[1] - pointB[1]);
  var squaredSum = Math.pow(xDiff, 2) + Math.pow(yDiff, 2);
  var unroundedSolution = Math.sqrt(squaredSum);
  var roundedSolution = Math.floor(unroundedSolution * 1000)/1000;
  console.log(roundedSolution); 
}

distance([0, 0], [0, 0])
distance([3, 0], [0, 0])
distance([0, 0], [0, -5])
distance([-3, 1], [0, 0])
